File: Product.py
class Product:
    def __init__(self, name, description, price, category):
        self.name = name
        self.description = description
        self.price = price
        self.category = category

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        if value is None:
            raise ValueError("Field cannot be empty. Type Required : Text")
        if not isinstance(value, str):
            raise TypeError("Name must be text")
        self._name = value

    @property
    def description(self):
        return self._description

    @description.setter
    def description(self, value):
        if value is None:
            raise ValueError("Field cannot be empty. Type Required : Text")
        if not isinstance(value, str):
            raise TypeError("Description must be text")
        self._description = value

    @property
    def price(self):
        return self._price

    @price.setter
    def price(self, value):
        if value is None:
            raise ValueError("Field cannot be empty. Type Required : Number")
        if not isinstance(value, (int, float)):
            raise TypeError("Price must be a positive number")
        if value < 0:
            raise ValueError("Price cannot be negative")
        self._price = value

    @property
    def category(self):
        return self._category

    @category.setter
    def category(self, value):
        if value is None:
            raise ValueError("Field cannot be empty. Type Required : List")
        if not isinstance(value, list):
            raise TypeError("Category must be a list of 1 or more")
        self._category = value

File: Backend/src/Store/Product.java
package Store;

import java.util.List;

public class Product {
    private String sku;
    private String name;
    private String description;
    private double price;
    private List<String> category;
    private int stock;

    public Product(String sku,
                   String name,
                   String description,
                   double price,
                   List<String> category,
                   int stock) {
        this.sku = sku;
        setName(name);
        setDescription(description);
        setPrice(price);
        setCategory(category);
        setStock(stock);
    }

    public String getSku() { return sku; }

    public String getName() { return name; }
    public void setName(String name) {
        if (name == null || name.trim().isEmpty()) {
            throw new IllegalArgumentException("Field cannot be empty. Type Required : Text");
        }
        this.name = name;
    }

    public String getDescription() { return description; }
    public void setDescription(String description) {
        if (description == null) {
            throw new IllegalArgumentException("Field cannot be empty. Type Required : Text");
        }
        this.description = description;
    }

    public double getPrice() { return price; }
    public void setPrice(double price) {
        if (Double.isNaN(price)) {
            throw new IllegalArgumentException("Field cannot be empty. Type Required : Number");
        }
        if (price < 0) {
            throw new IllegalArgumentException("Price cannot be negative");
        }
        this.price = price;
    }

    public List<String> getCategory() { return category; }
    public void setCategory(List<String> category) {
        if (category == null) {
            throw new IllegalArgumentException("Field cannot be empty. Type Required : List");
        }
        this.category = category;
    }

    public int getStock() { return stock; }

    public void setStock(int stock) {
        if (stock < 0) {
            throw new IllegalArgumentException("Stock cannot be negative");
        }
        this.stock = stock;
    }
}
